geowebcache/core/src/main/java/org/geowebcache/locks/NIOLockProvider.java

```diff
@@ -73,21 +84,22 @@ public NIOLockProvider(String root) throws ConfigurationException {
                 while(currLock == null && count < maxLockAttempts) {
                     // the file output stream can also fail to be acquired due to the
                     // other nodes deleting the file
-                    currFos = new FileOutputStream(file);
                     try {
+                        currFos = new FileOutputStream(file);
+
                         currLock = currFos.getChannel().lock();
                     } catch(OverlappingFileLockException e) {
                         IOUtils.closeQuietly(currFos);
```

Comment:
try-with-resource is not suitable solution here since the stream should not be closed if successfully created inside the try